### DISCUSSION TITLE
CustomEvent & x-on

### DISCUSSION THREAD

**Discussion by maubut-omt** [alpinejs/alpine#3910] [7. Bugs (archived)] [answered]
Hello,

I'm not sure why it's not working based on what I've read so far. My window.addEventListener does work, but not the @skillClicked.window. I am firing the event in my script.js file.

` const event = new CustomEvent("skillClicked", { detail: d.skill });
      window.dispatchEvent(event);`

..and in my html I have:

```
<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <script src="//unpkg.com/alpinejs" defer></script>
    <link rel="stylesheet" href="bundle.css" />

    <script>
      window.addEventListener("skillClicked", (event) => {
        console.log("Received skillClicked event:", event.detail);
      });
    </script>
  </head>
  <body>
    <div
      x-data="{ skill: null }"
      @skillClicked.window="console.log('Received event:', $event)"
    >
      <!-- ... -->
    </div>
...
```

What do I miss?

**Comment by hans-lts** [comment, marked as answer]
I've run into this before - it's because your event names are in [camel case](https://developer.mozilla.org/en-US/docs/Glossary/Camel_case). 

Alpine (specifically in this case `x-on:`) only supports lower-case event names. Hyphens are are though. 

Here's an experpt from the docs:

>x-on can only listen for events with lower case names, as HTML attributes are case-insensitive. Writing x-on:CLICK will listen for an event named click. If you need to listen for a custom event with a camelCase name, you can use the [.camel helper](https://alpinejs.dev/directives/on#camel) to work around this limitation. Alternatively, you can use [x-bind](https://alpinejs.dev/directives/bind#bind-directives) to attach an x-on directive to an element in javascript code (where case will be preserved).

_Source: https://alpinejs.dev/directives/on_

This is your example in a working state:

```html
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <script defer src="https://cdn.jsdelivr.net/npm/alpinejs@3.x.x/dist/cdn.min.js"></script>
    <script>
        // Added as a placeholder
        const d = {
            skill: "some data"
        }

        // Your custom event
        const event = new CustomEvent("skill-clicked", { detail: d.skill });

        // Fire the event once everything has loaded up
        window.addEventListener('DOMContentLoaded', () => {
            window.dispatchEvent(event);
        })
    </script>
</head>
<link rel="stylesheet" href="bundle.css" />
</head>

<body>
    <div x-data="{ skill: null }" @skill-clicked.window="console.log('Received event:', $event)">
        <!-- ... -->
    </div>
</body>

</html>
```
I updated the event name to `skill-clicked` but `skillclicked` would work too.

**Reply by ekwoka** [reply]
to clarify the above, this is because HTML does not support cased attributes. All attributes become lowercase in practice. SVG namespace has a view case sensitive attributes, but HTML does not.

**Reply by maubut-omt** [reply]
Thank you! It's working perfectly now :)
